### PR TITLE
WIP: added travis CI testing for prplmesh

### PR DESCRIPTION
**Pull request by slyusar003**
- added test param in maptools.py
- added check param in prplmesh-utils.py

Signed-off-by: Ostap <slyusar.ostap@gmail.com>

### CODE REVIEW

File: tools/maptools.py [hunk 1]
```diff
@@ -69,13 +72,17 @@ def __deploy__(self):
...
+
     def run(self):
         commands = {
             'config': self.__config__, 
             'build': self.__build__,
             'package': self.__package__,
             'deploy': self.__deploy__,
-            'copy': self.__copy__
+            'copy': self.__copy__,
+            'test': self.__test__,
```
Comment (arnout) [inline tools/maptools.py:85]: Great idea to add a `test` command to maptools.

Note that this should be in a separate commit from the changes to prplmesh_utils.sh and the changes to .travis.yml. So in the end, there should be 3 commits I think:

1. Add `check` to prplmesh_utils.sh
2. Add `test` to maptools
3. Add call to `test` to travis.yml

File: tools/maptools.py [hunk 2]
```diff
@@ -69,13 +72,17 @@ def __deploy__(self):
     def __copy__(self):
         mapcopy(self.args)
 
+    def __test__(self):
```
Comment (arnout) [inline tools/maptools.py:75]: Hm, it's wrong to make this a dunder method, since dunder methods are supposed to be reserved for the language. Still, the rest of the class does it like that, so for consistency it's good to do the same.

File: tools/commands/test.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import logging
+import argparse
+import os
+import subprocess
+
+logger = logging.getLogger("test")
+test_targets=['start', 'stop', 'restart', 'status', 'check']
+
+class prpltester(object):
```
Comment (arnout) [inline tools/commands/test.py]: I don't think it's useful to make this a separate class, just merge it into the maptest class.

Comment (slyusar003) [reply tools/commands/test.py]: okay, I was thinking that there will be more tests in the future but I will remove a class for now.

File: tools/commands/test.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    def run_command(self, tester, commands):
+        if 'start' in commands:
+            tester.run_cmd('start')
+        if 'stop' in commands:
+            tester.run_cmd('stop')
+        if 'check' in commands:
+            tester.run_cmd('restart')
+            tester.run_cmd('check')
+            tester.run_cmd('stop')
```
Comment (arnout) [inline tools/commands/test.py]: Originally I thought, better just do `for c in commands: tester.run_cmd(c)`. However, this really makes a lot of sense. :+1:

File: common/beerocks/scripts/prplmesh_utils.sh.in [hunk 5]
```diff
@@ -104,8 +104,13 @@ status_function() {
     ps -aux | grep local_bus
 }
 
+check_function() {
+    echo "$0: check"
+    cat "/tmp/$USER/beerocks/logs/beerocks_agent.log" |grep 'CONNECTED --> OPERATIONAL'
```
Comment (arnout) [inline common/beerocks/scripts/prplmesh_utils.sh.in]: You don't need `cat` here, just
```suggestion
    if ! grep -q 'CONNECTED --> OPERATIONAL' "/tmp/$USER/beerocks/logs/beerocks_agent.log"; then
        err "check failed"
        exit 1
    fi
```

File: common/beerocks/scripts/prplmesh_utils.sh.in [hunk 6]
```diff
@@ -104,8 +104,16 @@ status_function() {
     ps -aux | grep local_bus
 }
 
+check_function() {
```
Comment (tomereli) [inline common/beerocks/scripts/prplmesh_utils.sh.in]: I think it is better to add this into the status function

Comment (tomereli) [reply common/beerocks/scripts/prplmesh_utils.sh.in]: OPERATIONAL in the main agent doesn't mean the radio agents are operational - the status check should first indeed check that the main agent is operational, and then grep beerocks_agent_wlan0.log to see it got to a working state - for now, in DUMMY mode, we know the son slave is functional if we see `ACTION_AP_MANAGER_HOSTAP_CSA_NOTIFICATION`

File: common/beerocks/scripts/prplmesh_utils.sh.in [hunk 7]
```diff
@@ -104,8 +104,16 @@ status_function() {
...
+    echo "$0: check"
+    if ! grep -q 'CONNECTED --> OPERATIONAL' "/tmp/$USER/beerocks/logs/beerocks_agent.log"; then
+        err "check failed"
+        exit 1
+    fi
+}
+
 usage() {
-    echo "usage: $(basename $0) {start|stop|restart|status} [-hv]"
+    echo "usage: $(basename $0) {start|stop|restart|status|check} [-hv]"
```
Comment (tomereli) [inline common/beerocks/scripts/prplmesh_utils.sh.in]: no need for `check` - should be part of status, return code defines if system is operational, and we can add a --quiet flag if we want to get only the check status

**Review by tomereli — CHANGES_REQUESTED** [review]

File: tools/commands/test.py [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+    @staticmethod
+    def configure_parser(parser=argparse.ArgumentParser(prog='test')):
+        parser.help = "multiap_sw standalone test module"
+        parser.add_argument("--verbose", "-v", action="store_true", help="verbosity on")
+        parser.add_argument('-c', '--commands', choices=test_targets, nargs='+', default=['check'], help="test command (default is check)")
```
Comment (tomereli) [inline tools/commands/test.py]: I think a better name if you take my comments from above is to call this 'status'
`./maptools.py test -c status`

Then later we can add more tests, i.e. `./maptools.py test -c ctest`, etc.

File: common/beerocks/scripts/prplmesh_utils.sh.in (hunk 7 above)
Comment (arnout) [reply common/beerocks/scripts/prplmesh_utils.sh.in]: Requires more changes to status_function, however, because currently it doesn't set the exit code, nor does it use BTL_TYPE or PRPLMESH_MODE to determine which daemons should be running.
